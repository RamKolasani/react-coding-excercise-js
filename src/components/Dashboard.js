import React from 'react';
import DisplayMenuItems from './DisplayItems';


const Dashboard = (props) => {
    let MenuData= props.MenuData;
    return (
        <div className="dashboard">
                <ul className="items-list">
                    {Object.keys(MenuData).map(key =>{
                        return <li id={key}>
                                    <DisplayMenuItems item={MenuData[key]} />
                                </li>
                    })}
                </ul>
       </div>
    )
}

export default Dashboard;