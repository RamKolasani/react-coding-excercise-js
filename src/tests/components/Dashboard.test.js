import React from 'react';
import { shallow } from 'enzyme';
import Dashboard from '../../components/Dashboard';

describe('test Dashboard component',() =>{
    let wrapper;
    beforeAll(()=>{
        wrapper = shallow(<Dashboard MenuData={MenuData}/>)
    });

    test('test Dashboard snapshot',()=>{
        //const wrapper = shallow(<Dashboard MenuData={MenuData}/>);
        expect(wrapper).toMatchSnapshot();
    });
    
    test('should render Dashboard',()=>{
       // const wrapper = shallow(<Dashboard MenuData={MenuData}/>);
        expect(wrapper.find('div.dashboard')).toHaveLength(1);
    });

    test('should render 1 ul and 3 li items 3 3DisplayMenuItems',()=>{
        expect(wrapper.find('ul.items-list')).toHaveLength(1);
        expect(wrapper.find('li')).toHaveLength(3);
        expect(wrapper.find('DisplayMenuItems')).toHaveLength(3);
    } )
})



const MenuData = {
    "16306": {
      "name": "Coffee",
      "category": "Beverages",
      "itemId": 16306,
      "imageUrl": "https://presto-resources.s3.amazonaws.com/android__menueditor_images__/images/dynamic/Jw5kV6JUEeqSXYyFkDJJSQ.jpg",
      "price": 200,
      "tax": 8.25,
      "available": true,
      "description": "",
      "modifierGroups": {},
      "timePeriods": [
        {
          "price": null,
          "description": null,
          "daysOfWeek": [],
          "period": [],
          "alwaysAvailable": true
        }
      ]
    },
    "16307": {
      "name": "Sakura Sunrise",
      "category": "Beverages",
      "itemId": 16307,
      "imageUrl": "https://presto-resources.s3.amazonaws.com/android__menueditor_images__/images/dynamic/KIaecKJUEeqcuIyFkDJJSQ.jpg",
      "price": 400,
      "tax": 8.25,
      "available": true,
      "description": "Alcohol Free",
      "modifierGroups": {},
      "timePeriods": [
        {
          "price": null,
          "description": null,
          "daysOfWeek": [],
          "period": [],
          "alwaysAvailable": true
        }
      ]
    },
    "16308": {
      "name": "Shirley Temple",
      "category": "Beverages",
      "itemId": 16308,
      "imageUrl": "https://presto-resources.s3.amazonaws.com/android__menueditor_images__/images/dynamic/KlJfY6JUEeqPSYyFkDJJSQ.jpg",
      "price": 400,
      "tax": 8.25,
      "available": true,
      "description": "Alcohol Free",
      "modifierGroups": {},
      "timePeriods": [
        {
          "price": null,
          "description": null,
          "daysOfWeek": [],
          "period": [],
          "alwaysAvailable": true
        }
      ]
    }
}