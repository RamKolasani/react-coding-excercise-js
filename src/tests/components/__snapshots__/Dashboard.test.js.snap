// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Dashboard component test Dashboard snapshot 1`] = `
<div
  className="dashboard"
>
  <ul
    className="items-list"
  >
    <li
      id="16306"
    >
      <DisplayMenuItems
        item={
          Object {
            "available": true,
            "category": "Beverages",
            "description": "",
            "imageUrl": "https://presto-resources.s3.amazonaws.com/android__menueditor_images__/images/dynamic/Jw5kV6JUEeqSXYyFkDJJSQ.jpg",
            "itemId": 16306,
            "modifierGroups": Object {},
            "name": "Coffee",
            "price": 200,
            "tax": 8.25,
            "timePeriods": Array [
              Object {
                "alwaysAvailable": true,
                "daysOfWeek": Array [],
                "description": null,
                "period": Array [],
                "price": null,
              },
            ],
          }
        }
      />
    </li>
    <li
      id="16307"
    >
      <DisplayMenuItems
        item={
          Object {
            "available": true,
            "category": "Beverages",
            "description": "Alcohol Free",
            "imageUrl": "https://presto-resources.s3.amazonaws.com/android__menueditor_images__/images/dynamic/KIaecKJUEeqcuIyFkDJJSQ.jpg",
            "itemId": 16307,
            "modifierGroups": Object {},
            "name": "Sakura Sunrise",
            "price": 400,
            "tax": 8.25,
            "timePeriods": Array [
              Object {
                "alwaysAvailable": true,
                "daysOfWeek": Array [],
                "description": null,
                "period": Array [],
                "price": null,
              },
            ],
          }
        }
      />
    </li>
    <li
      id="16308"
    >
      <DisplayMenuItems
        item={
          Object {
            "available": true,
            "category": "Beverages",
            "description": "Alcohol Free",
            "imageUrl": "https://presto-resources.s3.amazonaws.com/android__menueditor_images__/images/dynamic/KlJfY6JUEeqPSYyFkDJJSQ.jpg",
            "itemId": 16308,
            "modifierGroups": Object {},
            "name": "Shirley Temple",
            "price": 400,
            "tax": 8.25,
            "timePeriods": Array [
              Object {
                "alwaysAvailable": true,
                "daysOfWeek": Array [],
                "description": null,
                "period": Array [],
                "price": null,
              },
            ],
          }
        }
      />
    </li>
  </ul>
</div>
`;
